Sources/PackageGraph/DependencyResolver.swift

```diff
@@ -433,7 +488,7 @@ struct VersionAssignmentSet<C: PackageContainer>: Equatable, Sequence {
         var result = PackageContainerConstraintSet<Container>()
         for (_, (container: container, binding: binding)) in assignments {
             switch binding {
-            case .excluded:
+            case .unversioned, .excluded:
```

Comment:
I think the constraints here should include those from the unversioned requirement, right?